stateMachine.go

```diff
@@ -0,0 +1,137 @@
+/*
```

Comment:
Small nit, but files are idiomatically named with snake case vs. camel case (ie. state_machine.go)

Reply:
Thanks, I'll make the change.